.github/workflows/build.yml

```diff
@@ -67,7 +67,7 @@ jobs:
         echo $DOCKER_IMG does not exist on dockerhub or pull failed
         echo This is normal for PR builds and for builds on forks
         echo Building from dockerfile
-        docker build tools/docker -t $DOCKER_IMG
+        docker build tools/docker -t $DOCKER_IMG --no-cache --pull
```

Comment:
Why do we need to add these parameters? Judging by the comment above, this code is hit when a pull failed. Why try `--pull` again?

Reply:
I just wanted to ensure no caching and force a pull from the base image too.
The pull was to enforce to get the latest image from the dockerhub.
The nocaching was to force the layer with the apt-get install to be ran. The layer will only run if the command in it or before was changed, however, this can leave the image with outdated libs and binaries.

This just ensures that we get the latest of everything. It might be an overkill but I see no harm in enforcing it.